src/lib/progress-bar/progress-bar.ts

```diff
@@ -61,13 +85,16 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements CanColor
                * @deprecated `location` parameter to be made required.
                * @breaking-change 8.0.0
                */
-              @Optional() location?: Location) {
+              @Optional() @Inject(MAT_PROGRESS_BAR_LOCATION) location?: MatProgressBarLocation) {
     super(_elementRef);
 
     // We need to prefix the SVG reference with the current path, otherwise they won't work
     // in Safari if the page has a `<base>` tag. Note that we need quotes inside the `url()`,
-    // because named route URLs can contain parentheses (see #12338).
-    this._rectangleFillValue = `url('${location ? location.path() : ''}#${this.progressbarId}')`;
+    // because named route URLs can contain parentheses (see #12338). Also we don't use
+    // `Location` from `@angular/common`, because we can't tell the difference between whether
+    // the consumer is using the hash location strategy or not.
```

Comment:
Couldn't you do the same thing with `location.path()` from `@angular/common`'s `Location`?

Reply:
That's what it used to do, but AFAIK Angular's `location.path` will normalize both of these to the same value: `/#/foo/bar` and `/foo/bar`.

Reply:
Ah, got it. Can you expand the comment with that? 
```
...the consumer is using the hash location strategy or not because it normalizes both
`/#/foo/bar` and `/foo/bar` to the same thing
```

Reply:
Done.